File: frontend/src/features/donor-management/services/fundProfileService.js
import { fundProfileApi } from '../api/fundProfileApi.js';
import {
  fromFundProfileResponse,
  toFundProfileRequest,
} from '../mappers/fundProfileMapper.js';

/**
 * Fund Profile domain service. Hooks/components call the service, never the
 * repository directly.
 */
export const fundProfileService = {
  async listByDonor(donorId) {
    const dtos = await fundProfileApi.listByDonor(donorId);
    return dtos.map(fromFundProfileResponse);
  },

  async getProfile(id) {
    return fromFundProfileResponse(await fundProfileApi.getById(id));
  },

  async createProfile(donorId, formValues) {
    return fromFundProfileResponse(
      await fundProfileApi.create(donorId, toFundProfileRequest(formValues)),
    );
  },

  async updateProfile(id, formValues) {
    return fromFundProfileResponse(
      await fundProfileApi.update(id, toFundProfileRequest(formValues)),
    );
  },
};

File: backend/finance/src/main/java/com/ngo/finance/donor/service/FundProfileService.java
package com.ngo.finance.donor.service;

import com.ngo.finance.donor.dto.request.CreateFundProfileRequest;
import com.ngo.finance.donor.dto.response.FundProfileResponse;
import java.util.List;

/**
 * Service for Donor Fund Profile operations (create / list-by-donor / get / update).
 */
public interface FundProfileService {

    FundProfileResponse createProfile(Long donorId, CreateFundProfileRequest request);

    List<FundProfileResponse> getProfilesByDonor(Long donorId);

    FundProfileResponse getProfileById(Long id);

    FundProfileResponse updateProfile(Long id, CreateFundProfileRequest request);
}
